src/scoping/langs/mod.rs

```diff
@@ -48,77 +55,61 @@ pub struct Language<Q> {
     /// type-safe (a [`TSQuery`] is *associated* with a [`TSLanguage`], but that's not
     /// enforced at the type level: a query constructed for Python could accidentally be
     /// passed to Go etc.), so use this field to be more type-safe.
-    _marker: PhantomData<Q>,
+    _marker: PhantomData<T>,
 }
 
-impl<Q> Language<Q>
-where
-    Q: Into<TSQuery> + Clone,
-{
+impl<T: Kind> Language<T> {
```

Comment:
Could this be simpler? Paraphrasing (might not work!):

```rust
impl<T: Into<TSLanguage>> Language<T> {
```

where `TSLanguage` is:

https://github.com/tree-sitter/tree-sitter/blob/079c69313fa14b9263739b494a47efacc1c91cdc/lib/binding_rust/lib.rs#L61

and then we can just pass e.g. `tree_sitter_rust::LANGUAGE` into that, which implements it via:

https://github.com/tree-sitter/tree-sitter/blob/079c69313fa14b9263739b494a47efacc1c91cdc/lib/binding_rust/lib.rs#L423

then inside of `new` can call `T.into()` (might need turbofish?) instead of `&T::ts_lang()`.

Advantages I have in mind: saves the roundtrip through `LangKind` structs, and no `Kind` trait necessary.

But again, just a quick idea which might not work.

And note: sadly, HCL is vendored here as it's not published as a crate properly. I'm open to changing that! I can look into publishing a `srgn-tree-sitter-hcl` crate, so all languages can be treated the same (HCL is currently the odd one out and slightly different, which is annoying).

Reply:
Fair point. I'll look into an `<T: Into<TSLanguage>>` based solution.

Reply:
I've made some quite large changes in https://github.com/alexpovel/srgn/pull/144/commits/d6fc6af3e724178a6f262ad8b27d98a7ca5c37ce.

I'll try and explain my changes below. Please see this commit as a proposal rather than me trying to demand changes.

I think that the `PhantomDatd` field in `scoping::langs::Language` is causing more trouble than it's worth.
The generic param `Q` can't be fulfilled by any of the `tree-sitter` language types as  they are not a struct unique to the given language. The type `tree-sitter::LanguageFn` can't be used as a generic param as it's not a struct or enum. So you end up needing to define a local struct per language mod to be used for the `Q` param.

I opted to just manually define a 'struct Language' per language mod instead of defining an alias of a `PhantomData` typed struct. So in short I replaced the generic `Language<Q>` with a per mod struct that composes a non-generic `Language`. 

This commit is made more complex by renaming the `struct Language` to a name I consider more representitive of how it's being used. I  propose renaming `struct Language<Q>` to `struct CompiledQuery`. 
Compiled query as opposed to a `RawQuery`.

This renaming makes the code simpler to grok for me at least.
See what you think yourself.

Reply:
I commented on this a bit already, but sounds good to me!

I had introduced the struct with `PhantomData` to be extra strict, and extra neat, but it never played well with the `tree_sitter` API (leaky abstraction), and ended up confusing, while not helping as you note.